Src/Plugins/Security/SEC101_013.CryptographicPrivateKeyValidator.cs

```diff
@@ -128,38 +128,50 @@ protected override IEnumerable<ValidationResult> IsValidStaticHelper(Dictionary<
 
         private static ValidationState GetPrivatePgpKey(string secret)
         {
-            using Stream keyIn = new MemoryStream(Encoding.UTF8.GetBytes(secret));
-            using Stream stream = PgpUtilities.GetDecoderStream(keyIn);
-            var secretKeyRingBundle = new PgpSecretKeyRingBundle(stream);
+            try
+            {
+                using Stream keyIn = new MemoryStream(Encoding.UTF8.GetBytes(secret));
+                using Stream stream = PgpUtilities.GetDecoderStream(keyIn);
+                var secretKeyRingBundle = new PgpSecretKeyRingBundle(stream);
 
-            bool oneOrMorePassphraseProtectedKeys = false;
+                bool oneOrMorePassphraseProtectedKeys = false;
 
-            foreach (PgpSecretKeyRing kRing in secretKeyRingBundle.GetKeyRings())
-            {
-                foreach (PgpSecretKey secretKey in kRing.GetSecretKeys())
+                foreach (PgpSecretKeyRing kRing in secretKeyRingBundle.GetKeyRings())
                 {
-                    PgpPrivateKey privateKey = null;
-                    try
+                    foreach (PgpSecretKey secretKey in kRing.GetSecretKeys())
                     {
-                        char[] noPassphrase = Array.Empty<char>();
-                        privateKey = secretKey.ExtractPrivateKey(noPassphrase);
-                    }
-                    catch (PgpException)
-                    {
-                        oneOrMorePassphraseProtectedKeys = true;
-                        continue;
+                        PgpPrivateKey privateKey = null;
+                        try
+                        {
+                            char[] noPassphrase = Array.Empty<char>();
+                            privateKey = secretKey.ExtractPrivateKey(noPassphrase);
+                        }
+                        catch (PgpException)
+                        {
+                            oneOrMorePassphraseProtectedKeys = true;
+                            continue;
+                        }
+
+                        return ValidationState.Authorized;
                     }
+                }
 
-                    return ValidationState.Authorized;
+                if (oneOrMorePassphraseProtectedKeys)
+                {
+                    return ValidationState.PasswordProtected;
                 }
-            }
 
-            if (oneOrMorePassphraseProtectedKeys)
-            {
-                return ValidationState.PasswordProtected;
+                return ValidationState.NoMatch;
             }
+            catch (Exception ex)
+            {
+                if (ex.Message == "unknown object in stream Reserved")
```

Comment:
>unknown object in stream Reserved [](http://example.com/codeflow?start=35&length=33)

for some malformed strings, it throws this exception. Let's catch and return noMatch.